docs/engineering.md

```diff
@@ -0,0 +1,595 @@
+```
+
+## Viewing Sent Telemetry Probes
+
+### client
+
+1. **Use the QA Helper Add-on**
+
+   The QA-Shield-Study-Helper lists the `payload.data` field for every `shield-study` and `shield-study-addon` ping.
+
+   [Bugzilla for QA Helper Add-on](https://bugzilla.mozilla.org/show_bug.cgi?id=1407757)
+   [direct install link for Signed XPI for @qa-shield-study-helper-1.0.0.xpi][qa-helper-addon-direct]
+
+   Example output:
+
+   ```text
+   // common fields
+
+   branch        up-to-expectations-1        // should describe Question text
+   study_name    57-perception-shield-study
+   addon_version 1.0.0
+   version       3
+   ```
+
+
+    2017-10-09T14:16:18.042Z shield-study
+    {
+      "study_state": "enter"
+    }
+    2017-10-09T14:16:18.055Z shield-study
+    {
+      "study_state": "installed"
+    }
+    2017-10-09T14:16:18.066Z shield-study-addon
+    {
+      "attributes": {
+        "event": "prompted",
+        "promptType": "notificationBox-strings-1"
+      }
+    }
+    2017-10-09T16:29:44.109Z shield-study-addon
+    {
+      "attributes": {
+        "promptType": "notificationBox-strings-1",
+        "event": "answered",
+        "yesFirst": "1",
+        "score": "0",
+        "label": "not sure",
+        "branch": "up-to-expectations-1",
+        "message": "Is Firefox performing up to your expectations?"
+      }
+    }
+    2017-10-09T16:29:44.188Z shield-study
+    {
+      "study_state": "ended-neutral",
+      "study_state_fullname": "voted"
+    }
+    2017-10-09T16:29:44.191Z shield-study
+    {
+      "study_state": "exit"
+    }
+    ```
+
+2. Use `about:telemetry`, and look for `shield-study` or `shield-study-addon` probes.
+
+### Collector (example s.t.m.o query)
+
+[Example s.t.m.o study states query for "Pioneer Enrollement"][stmo-study-states] shows the Study lifecycle for every client in the Pioneer Enrollment study.
+
+## Engineering Side-by-Side (a/b) Feature Variations
+
+Note: this is a gloss / summary.
+
+1. Your feature has a `startup` or `configuration` method that does different things depending on which variation is chosen.
+
+   ```javascript
+   // bootstrap.js startup()...
+   const variation = await studyUtils.deterministicVariation(myWeightedVariations);
+   studyUtils.setVariation(variation);
+
+   //...
+
+   // start the feature
+   TheFeature.startup(variation)
+   ```
+
+2. Ensure that your Feature measures every variation, including the Control (no-effect).
+
+## Kittens or Puppers, the Critical Study We have all been waiting for
+
+Style:
+
+* Embedded Web Extension
+* Telmetry on 'button click'
+* has one "end early" condition: 3 or more button presses during a sesson.
+* Goal: test if 'interest rate is higher for kittens or puppies, using a PROXY MEASURE -- "button clicks"
+
+## Get More Help
+
+* slack: `#shield`
+
+## Gotchas / FAQ / Ranting
+
+### General
+
+I am on Windows. How can I build?
+
+* (see TODO link to the issue and instructions by JCrawford)
```

Comment:
nit: needs link (which I guess is the point of the TODO). I searched open and closed issues/PRs and couldnt find mentions of Windows.